### ISSUE TITLE
Improve GKE deployment examples

### ISSUE THREAD

**Issue by davecheney** [projectcontour/contour#946]
We need better documentation for how to recover the remote IP address details when Contour is deployed behind GKE. See #945

**Comment by davecheney** [comment]
@youngnick do you think you can wrap this into #1349

**Comment by youngnick** [comment]
Hmm, missed this, but I'll take it and make it happen.

**Comment by davecheney** [comment]
Thanks

**Comment by youngnick** [comment]
I've investigated this over the last couple of days, here's what I've tried and found:

- I played around with using `--enable-proxy-protocol`, but this doesn't work. It doesn't work because when you create a Service of type LoadBalancer in GKE, you get a network load balancer, not a HTTP or TCP one. So it forwards packets unaltered to the Envoy instances that Contour configures.
- Changing the state of the Envoy HTTP Connection manager property UseRemoteAddress (which can sometimes tell Envoy to use the X-Forwarded-For header) from `true` to `false` means that you get the local node's IP address, even with `externalTrafficPolicy: local` set on the service.

In short, if you're doing something close to what our example deployment suggests on GKE, you will be able to retrieve the remote client IP address from the `X-Envoy-External-Address` request header.
 
I'll have a crack at adding some documentation around this.
